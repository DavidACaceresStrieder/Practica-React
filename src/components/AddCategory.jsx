import { useImperativeHandle } from "react";
import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => 
{
    const [ inputValue , setInputValue ] = useState('');    

    const onInputChanged = (event) => setInputValue(event.target.value); 
    const onSubmitForm = (event) => 
    {   
        event.preventDefault();        

        const value = inputValue.trim();
        if(value.length == 0) return;

        onNewCategory( value );
        setInputValue('');
    }

    return(
        <>
        <form onSubmit={ onSubmitForm } aria-label="form">
            <input
                type="text"
                placeholder="Buscar Gifs"
                value = {inputValue}
                onChange={ (event) => onInputChanged(event) }
            >
            </input>
        </form>
        
        </>
    );
}

AddCategory.proptypes = {
    onNewCategory : PropTypes.func.isRequired
}