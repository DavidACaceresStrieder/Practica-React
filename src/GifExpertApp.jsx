import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () =>
{
    const [ categories, setCategories ] = useState([]);
    
    const onAddCategory = (value) => 
    {
        if(categories.includes(value)) return;
        setCategories([...categories , value]);
    };

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory setCategories={ setCategories } onNewCategory = { (value) => onAddCategory(value) } />            
            {  
                categories.map( category => 
                { 
                    return ( <GifGrid key={category} category={category}/> )                
                }) 
            }
            
        </>
    );
}