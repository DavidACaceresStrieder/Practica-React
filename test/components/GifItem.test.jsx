import { GifItem } from "../../src/components";
import { render } from '@testing-library/react';

describe('Pruebas del <GifItem />', () =>
{
    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';

    test('Debe hacer match con el snapshot', () => 
    {
        const { container } = render( <GifItem  title= {title} url={url} ></GifItem>);
        expect(container).toMatchSnapshot();
    });
})