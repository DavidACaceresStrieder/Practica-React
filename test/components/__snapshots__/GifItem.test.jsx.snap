// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
       
      Saitama
       
    </p>
  </div>
</div>
`;
